src/middlewared/middlewared/plugins/system.py

```diff
@@ -1359,11 +1384,47 @@ def set_crash_reporting(self):
         CrashReporting.enabled_in_settings = self.middleware.call_sync('system.general.config')['crash_reporting']
 
 
+async def _update_birthday_data(middleware, birthday=None):
+    if birthday is None:
```

Comment:
This method should not set the birthday if it is already set (loop from `_update_birthday` won't be aborted if `'interface.post_sync'`, and next iteration will overwrite birthday value that has already been set)